x-pack/plugins/security_solution/public/flyout/index.tsx

```diff
@@ -90,8 +90,30 @@ const expandableFlyoutDocumentsPanels: ExpandableFlyoutProps['registeredPanels']
   },
 ];
 
+/**
+ * Flyout used for the Security Solution application
+ * We set the z-index to 999 to ensure it is always rendered behind Timeline
+ */
 export const SecuritySolutionFlyout = memo(() => (
-  <ExpandableFlyout registeredPanels={expandableFlyoutDocumentsPanels} paddingSize="none" />
+  <ExpandableFlyout
+    registeredPanels={expandableFlyoutDocumentsPanels}
+    paddingSize="none"
+    customStyles={{ 'z-index': 999 }}
+  />
 ));
 
 SecuritySolutionFlyout.displayName = 'SecuritySolutionFlyout';
+
+/**
+ * Flyout used in Timeline
+ * We set the z-index to 1001 to ensure it is always rendered above Timeline
+ */
+export const TimelineFlyout = memo(() => (
+  <ExpandableFlyout
+    registeredPanels={expandableFlyoutDocumentsPanels}
+    paddingSize="none"
+    customStyles={{ 'z-index': 1001 }}
```

Comment:
The timeline flyout will cover flyouts such as `Add to new case`, `Add Endpoint exception` and `Add rule exception` from take action. The new case flyout will stay after closing the timeline flyout. But endpoint and rule exception are not usable, because they always close automatically when timeline flyout is closed. 

https://github.com/elastic/kibana/assets/18648970/f1926861-e119-44be-82c7-cb73f529aa75

Reply:
crap, I thoughts had looked at some of these and their `z-index` was higher (I remember seeing on at like `8000` lol)...
Thanks for testing this, I'll look into it!

Reply:
very interesting. I had tested this for the `eventFlyout` and it was working correctly. I saw that those have a custom `z-index` (set to `8000` for the Create case flyout for example) so I did not worry about it.
I'm not sure why it's not working correctly for the `timelineFlyout` as the `z-index: 8000` is there...
![Screenshot 2024-02-21 at 5 10 37 PM](https://github.com/elastic/kibana/assets/17276605/920c9cfe-a396-4740-90b3-9068fd675227)

Reply:
ok after more investigation, the reason this is happening is because all of these flyouts opened from the `Take action` (Create new case, Add Endpoint exception, Add rule exception...) are using an `overlayMask` from EUI. This which adds a class to the very top parent of those flyout which sets a `z-index` to 1000. Internally they set a `z-index` to 8000 to the actual flyout component, but that doesn't matter as a parent has another `z-index` set. I was confused at first by that.

I decided to - instead of making changes to all of these flyouts that we don't own and would be used in other places we're not aware of - make a small change to the `z-index` values we can control:
- the `eventFlyout` `z-index` value is now `997` instead of `999`
- the `timeline` `z-index` value is now `998` instead of `1000`
- the `timelineFlyout` `z-index` value is now `999` instead of `1000`

This should guarantee that these are still playing nice with each other, while staying behind the other `Take action` flyouts or other potential modals being open...